### DISCUSSION TITLE
Zot validates to empty value if a select option is selected

### DISCUSSION THREAD

**Discussion by Stefan4691** [colinhacks/zod#2921] [Q&A]
Hi devs.

If I build an select with zot, zot always claims it as empty.

`const categories = ["Privat", "Firma Office", "Firma Projekte"] as const;`

```
const schema = z.object({
  category: z.enum(categories)
});
```
`type TodoFormData = z.infer<typeof schema>;`

      <div className="mb-3">
        <label htmlFor="category" className="form-label">
          Category
        </label>
        <select
          {...register("category")}
          name="Category"
          id="category"
          className="form-select"
          onChange={(obj) => console.log(obj.target.value)}
        >
          <option value=""></option>
          {categories.map((category) => (
            <option key={category} value={category}>
              {category}
            </option>
          ))}
        </select>
        {errors.category && (
          <p className="text-danger">{errors.category.message}</p>
        )}
      </div>
 
If I change the values in the select field, my 'onChange' prints the new value correctly.
Submitting the form, the category is empty

<img width="247" alt="image" src="https://github.com/colinhacks/zod/assets/44541610/5e734777-2528-4170-84ec-bbdf2b925e3b">

I also tried to set the schema to

`category: z.string()
`
without luck.

I can't see the error... any idea?
Thanks

**Comment by Stefan4691** [comment]
I figured that it not evaluates to an empty select value in general, but to the option which was selected at the first rendering.

MEans if I set   

```
<option value="myoption">My Option</option>
      {categories.map((category) => (
        <option key={category} value={category}>
          {category}
        </option>
```

it evaluates to 'myoption' regardless what option I was selecting.

**Reply by SharmaTushar** [reply]
Please have a look at #2466
